### ISSUE TITLE
LazyLoad a module from javascript directly

### ISSUE THREAD

**Issue by gvandyk** [ocombe/ocLazyLoad#165]
I want to write a component/module that is totally self contained and I don't want to change code in other modules/controllers to load my module.

Currently I include a javascript file at the bottom of an html file that defines all the logic, controllers, components, directives etc.

How can I LazyLoad this file, without having to update another controller somewhere.

for instance: LazyLoadModule('mymodule.js'), instead of having the $ocLazyLoad service available.

Using the oc-lazy-load directive works perfectly well, if I surround all my html with a div that includes that directive. However I also need to write components with which I cannot do this.

So, in short: How can I call LazyLoad to load my module directly from within Javascript.

**Comment by ocombe** [comment]
Hmm I'm not sure if I understand correctly what you're trying to do.
When you say "from javascript" you mean from javascript that is not in an angular application ?
Or are you trying to make your module auto load itself just by including the file ? If that's the case, will ocLazyLoad be available in your main app?

**Comment by gvandyk** [comment]
Yes, the javascript is not an angular application, but ocLazyLoad is available in the mainapp.

I just can't figure out howto call lazyload externally from the "app".

**Comment by ocombe** [comment]
You could use angular.injector: https://docs.angularjs.org/api/ng/function/angular.injector
With it you can get any service available in your application.

**Comment by gvandyk** [comment]
Thanks Olivier

I tried the following:

myMod.html
```
1: <div oc-lazy-load="['/static/js/nux/workflow/taskList.js']">
2: 	<div ng-controller='TaskListPageController as ctrl'>
3:		{% verbatim %}
4:			<span>{{ctrl.name}}</span>
5:		{% endverbatim %}
6:	</div>
7:</div>
8:
9:<script>
10:var injector = angular.element(document.body).injector();	
11:
12:injector.invoke(function($ocLazyLoad) {
13:	$ocLazyLoad.load(['/static/js/nux/workflow/taskList.js']);
14:});	
15:</script>
```

taskList.js
```
angular.module('myApp')
	.controller('TaskListPageController',['$scope', function($scope) {
		this.name = 'Testing Controller worked';
	}]);
```

When I comment out line:

**Comment by ocombe** [comment]
Yes, the load function is asynchronous, and besides your script is at the end of the DOM so the DOM will be evaluated first.

You might have to disable the DOM using ng-if or ng-include based on a scope variable that you enable once the loading has been done.

Something like this:
```html
<div ng-if="ctrlLoaded" ng-controller='TaskListPageController as ctrl'>
    {% verbatim %}
        <span>{{ctrl.name}}</span>
    {% endverbatim %}
</div>

<script>
var injector = angular.element(document.body).injector();    

injector.invoke(function($ocLazyLoad, $rootScope) {
 $ocLazyLoad.load(['/static/js/nux/workflow/taskList.js']).then(function() {
    $rootScope.ctrlLoaded = true;
  }, function(err) {
    console.log(err);
  });
});  
</script>
```
